### ISSUE TITLE
Connector can't renew properly Kerberos ticket

### ISSUE THREAD

**Issue by AlexPiermatteo** [confluentinc/kafka-connect-hdfs#52]
Hi,

I've this problem in a Kerberized environment: when I start the connector everything is working fine, I obtain my Kerberos credentials and the connector start writing without issues. The problem begin when a day after the ticket from Kerberos is renewed and the connector crash immediatly with this error:

> 
ERROR Recovery failed at state RECOVERY_PARTITION_PAUSED (io.confluent.connect.hdfs.TopicPartitionWriter:221) org.apache.kafka.connect.errors.ConnectException: java.io.IOException: Failed on local exception: java.io.IOException: javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: Failed to find any Kerberos tgt)]; Host Details : local host is: "mitstatlodpbroker01/10.72.176.22"; destination host is: "mitstatlodpmaster01":8020;
        at io.confluent.connect.hdfs.wal.FSWAL.apply(FSWAL.java:131)
        at io.confluent.connect.hdfs.TopicPartitionWriter.applyWAL(TopicPartitionWriter.java:519)
        at io.confluent.connect.hdfs.TopicPartitionWriter.recover(TopicPartitionWriter.java:204)
        at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:234)
        at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:234)
        at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:91)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:287)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:176)
        at org.apache.kafka.connect.runtime.WorkerSinkTaskThread.iteration(WorkerSinkTaskThread.java:90)
        at org.apache.kafka.connect.runtime.WorkerSinkTaskThread.execute(WorkerSinkTaskThread.java:58)
        at org.apache.kafka.connect.util.ShutdownableThread.run(ShutdownableThread.java:82)

> 

it is strange because the connector is able to renew periodically the ticket (due to the async function in DataWriter class) but the issue still there and I've to restart manually the connector every time that this error happen.

Did you know this error or there is something that maybe I'm doing wrong?

Thanks.

**Comment by AlexPiermatteo** [comment]
Hi,

after some researches I found the problem, is related to the hadoop-common library 2.6.0 and Java 8. 
Basically the function UGI#reloginFromKeytab inside the DataWriter.java class used to renew everyday the Kerberos ticket fail silently due to a parameter (isKeyTab) that returns false every time even if should be true (link bug: https://issues.apache.org/jira/browse/HADOOP-10786).

I solved the issue using Java 7 to run the connector and also with the usage of Hadoop-Clent 2.6.1 library that has the fix inside.

Regards,

Alex
